### PR TITLE
Core/Achievements: fixed bug with counting honorless units for achievement type ACHIEVEMENT_CRITERIA_TYPE_SPECIAL_PVP_KILL

### LINKED ISSUE #16075: [3.3.5][6.x] Core/Achievement: City Defender achievement awarding random kills
The achievement City Defender (ID:1006)
is bugged. Something is awarding kills when it shouldn't.

I think it awards kills when you kill mind controlled allies in instances.

rev. e6903fdffa73 2015-12-15

### PR DESCRIPTION
**Pull request by ForesterDev**
**Target branch(es):** 3.3.5/master

- [x] 3.3.5

**Issues addressed:** Closes #16075

**Tests performed:** (Does it build, tested in-game, etc.)
Build, tested

### CODE REVIEW

**Comment by jackpoz** [comment]
> need tests

@ForesterDev test it please (and debug it line by line while at it)

File: src/server/game/Entities/Player/KillRewarder.cpp [hunk 1]
```diff
@@ -232,17 +232,9 @@ void KillRewarder::_RewardGroup()
 
             // 3.1.3. Reward each group member (even dead or corpse) within reward distance.
             for (GroupReference* itr = _group->GetFirstMember(); itr != nullptr; itr = itr->next())
-            {
```
Comment (jackpoz) [inline src/server/game/Entities/Player/KillRewarder.cpp]: please revert all the codestyle changes, they make it harder to review what was actually changed

**Review by jackpoz — CHANGES_REQUESTED** [review]

**Comment by ForesterDev** [comment]
Just tested ... only honorable kills, when victim is player, count in achievement "City Defender"

File: src/server/game/Entities/Player/KillRewarder.cpp (hunk 1 above)
Comment (ForesterDev) [reply src/server/game/Entities/Player/KillRewarder.cpp]: But this brackets become unnecessary with line removal

Comment (illfated) [reply src/server/game/Entities/Player/KillRewarder.cpp]: Maybe keep at least the `for` loop brackets, since there is more than 1 line inside that loop.

**Comment by jackpoz** [comment]
I ended up removing the codestyle changes myself...

**Comment by jackpoz** [comment]
Thanks for the PR :)

**Pull request merged**